include/dem/from_file_insertion.h

```diff
@@ -0,0 +1,108 @@
+/* ---------------------------------------------------------------------
+ *
+ * Copyright (C) 2019 - 2019 by the Lethe authors
+ *
+ * This file is part of the Lethe library
+ *
+ * The Lethe library is free software; you can use it, redistribute
+ * it, and/or modify it under the terms of the GNU Lesser General
+ * Public License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE at
+ * the top level of the Lethe distribution.
+ *
+ * ---------------------------------------------------------------------
+ *
+ */
+
+#include <core/dem_properties.h>
+
+#include <dem/dem_solver_parameters.h>
+#include <dem/insertion.h>
+
+#include <deal.II/distributed/tria.h>
+
+#include <deal.II/particles/particle_handler.h>
+
+#ifndef lethe_from_file_insertion_h
+#  define lethe_from_file_insertion_h
+
+template <int dim>
+class FromFileInsertion : public Insertion<dim>
+{
+public:
+  FromFileInsertion(const DEMSolverParameters<dim> &dem_parameters,
+                    const std::vector<std::shared_ptr<Distribution>>
+                      &distribution_object_container);
+
+  /**
+   * @brief The FromFileInsertion class inserts particles using data stored in a file.
+   * This allows the insertion of any number of particles at a well-controlled
+   * location with any diameter value, translation and angular velocity.
+   *
+   * @param particle_handler The particle handler of particles which are being
+   * inserted
+   * @param triangulation Triangulation to access the cells in which the
+   * particles are inserted
+   * @param dem_parameters DEM parameters declared in the .prm file
+   */
+  virtual void
+  insert(Particles::ParticleHandler<dim>                 &particle_handler,
+         const parallel::distributed::Triangulation<dim> &triangulation,
+         const DEMSolverParameters<dim> &dem_parameters) override;
+
+
+  /**
+   * @brief Carries out assigning the properties of inserted particles specifically
+   * for the from_file insertion method. In this method, the initial translation
+   * and angular velocities and the diameter of each particles is set.
+   *
+   * @param dem_parameters DEM parameters declared in the .prm file
+   * @param inserted_this_step_this_proc Number of particles that are inserted
+   * at each insertion step on each processor. This value can change in the last
+   * insertion step to reach the desired number of particles
+   * @param particles_data Contains the particles
+   * @param particle_properties Properties of all inserted particles at each insertion step
```

Comment:
```suggestion
   * @param dem_parameters DEM parameters declared in the .prm file
   
   * @param inserted_this_step_this_proc Number of particles that are inserted
   * at each insertion step on each processor. This value can change in the last
   * insertion step to reach the desired number of particles
   
   * @param particles_data Contains the particles
   
   * @param particle_properties Properties of all inserted particles at each insertion step
```

Reply:
Why those spaces? It makes files a bit heavy if we do that for all files, no?

Reply:
I don't know, I thought this was the standard approach we had agreed on 
All the other files in core and solvers seem to follow that standard
e.g. https://github.com/lethe-cfd/lethe/blob/master/include/core/bdf.h